obsah/data_types.py

```diff
@@ -0,0 +1,114 @@
+"""
+obsah data types to be used to validate user input
+"""
+
+import argparse
+import os
+import pathlib
+import re
+import sys
+
+REGISTRY: list[type['BaseType']] = []
```

Comment:
I think this typing only works on fairly modern Python

Reply:
I already wrote https://github.com/theforeman/obsah/pull/64 but then thought the benefit would be small. Now submitted as a PR.